contracts/multisig/src/key.rs

```diff
@@ -1,55 +1,113 @@
+use crate::{secp256k1::ecdsa_verify, types::MsgToSign, ContractError};
 use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{HexBinary, StdError, StdResult};
 use cw_storage_plus::{KeyDeserialize, PrimaryKey};
+use enum_display_derive::Display;
 use serde::{de::Error, Deserializer};
-use std::marker::PhantomData;
-
-use crate::{secp256k1::ecdsa_verify, types::MsgToSign, ContractError};
+use std::fmt::Display;
 
 #[cw_serde]
-#[derive(Copy)]
+#[derive(Copy, Display)]
 pub enum KeyType {
     Ecdsa,
 }
 
 #[cw_serde]
 #[derive(PartialOrd, Ord, Eq)]
-pub enum Signature<T>
-where
-    T: Recoverability,
-{
-    Ecdsa(HexBinary, PhantomData<T>),
+pub enum Signature {
+    Ecdsa(NonRecoverable),
+    EcdsaRecoverable(Recoverable),
```

Comment:
I don't think we need this. I think this still creates confusion. I think we should just have Ecdsa, and the signatures are always non-recoverable.

Reply:
yeah, recoverable is just an encoding format for ecdsa signatures, not a different type of signature scheme